src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -195,7 +423,58 @@ int CPUPerformanceInterface::CPUPerformance::cpu_loads_process(double* pjvmUserL
 
   return result;
 #else
-  return FUNCTIONALITY_NOT_IMPLEMENTED;
+  JVMTicks curJVMTicks, *prevJVMTicks;
+  CPUTicks *curCPUTicks, *prevCPUTicks;
+
+  uint64_t jvmUserDelta, jvmSystemDelta, cpuUsedDelta, cpuTotalDelta;
+
+  *pjvmUserLoad = 0.0;
+  *pjvmKernelLoad = 0.0;
+  *psystemTotalLoad = 0.0;
+
+  if (_counters.cpus == NULL) {
+    return OS_ERR;
+  }
+
+  if (get_jvm_ticks(&curJVMTicks, _counters.stathz, _counters.nProcs) != OS_OK) {
+    return OS_ERR;
+  }
+
+  prevJVMTicks = &_counters.jvmTicks;
+  curCPUTicks = &curJVMTicks.cpuTicks;
+  prevCPUTicks = &prevJVMTicks->cpuTicks;
+
+  jvmUserDelta = curJVMTicks.userTicks > prevJVMTicks->userTicks ?
```

Comment:
Can it ever be true that the current ticks are not >= the previous value?  I'm just trying to understand if this is a real scenario we're guarding against.

Reply:
I thought I replied to this. Sorry if this is duplicate. I saw this in the linux implementation:
```
  // seems like we sometimes end up with less kernel ticks when
  // reading /proc/self/stat a second time, timing issue between cpus?
```
So I figured I would have this bit of defensive programming to cover any unexpected values returned by the OS. I did not see a need for this when testing on OpenBSD.